x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/dataframe/DataFrameAnalysisConfig.java

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+package org.elasticsearch.xpack.core.ml.dataframe;
+
+import org.elasticsearch.common.io.stream.StreamInput;
+import org.elasticsearch.common.io.stream.StreamOutput;
+import org.elasticsearch.common.io.stream.Writeable;
+import org.elasticsearch.common.xcontent.ContextParser;
+import org.elasticsearch.common.xcontent.ToXContentObject;
+import org.elasticsearch.common.xcontent.XContentBuilder;
+import org.elasticsearch.xpack.core.ml.utils.ExceptionsHelper;
+
+import java.io.IOException;
+import java.util.Map;
+import java.util.Objects;
+
+public class DataFrameAnalysisConfig implements ToXContentObject, Writeable {
```

Comment:
you could add `extends AbstractDiffable<DataFrameAnalysisConfig>` but I wonder whether that makes a big difference, I guess `AbstractDiffable` saves bandwidth when the config is stored in cluster state, but wonder whether it is used if it isn't, assuming you don't use cluster state.

Reply:
Yes, the diffable framework makes sense for things stored in the cluster state. I'd rather not implement it until needed.